src/Language/PureScript/Linter/Exhaustive.hs

```diff
@@ -258,13 +255,14 @@ checkExhaustive env mn numArgs cas = makeResult . first nub $ foldl' step ([init
 
   makeResult :: ([[Binder]], (Either RedudancyError Bool, [[Binder]])) -> m ()
   makeResult (bss, (rr, bss')) =
-    do unless (null bss) tellExhaustive
+    do unless (hasConstraint || null bss) tellNonExhaustive
        unless (null bss') tellRedundant
        case rr of
-         Left Incomplete -> tellIncomplete
-         _ -> return ()
+         Left Incomplete -> unless hasConstraint tellIncomplete
+         _ -> when (hasConstraint && null bss) tellExhaustive
     where
-    tellExhaustive = tell . errorMessage . uncurry NotExhaustivePattern . second null . splitAt 5 $ bss
+    tellExhaustive = tell . errorMessage $ NotPartial
```

Comment:
What about

```haskell
loop :: forall a b. Partial => a -> b
loop a = loop a
```

? I'd call that `Partial`. I don't think the class should be restricted to one particular type of use.

Or what if the user wants to propagate a `Partial` constraint to an otherwise total caller?